editoast/src/views/mod.rs

```diff
@@ -618,26 +618,31 @@ async fn authenticate(
         }
         (None, _) => return Ok(Authentication::Unauthenticated),
         (Some(identity), name) => UserInfo {
-            identity,
+            identities: vec![identity],
             name: name.unwrap_or_default(),
         },
     };
 
-    let authorizer =
-        match Authorizer::try_initialize(user.identity.clone(), regulator.clone()).await {
-            Ok(authorizer) => authorizer,
-            Err(AuthorizerError::UnknownUser { .. }) => {
-                // The user is not in the database, let's add it
-                regulator
-                    .clone()
-                    .driver()
-                    .ensure_user(&user)
-                    .await
-                    .map_err(AuthorizerError::Storage)?;
-                Authorizer::try_initialize(user.identity, regulator.clone()).await?
-            }
-            Err(err) => return Err(err.into()),
-        };
+    let authorizer = match Authorizer::try_initialize(
+        identity.clone().expect("Already checked above"),
```

Comment:
Perhaps the code above should return a tuple of `(user, identity)` instead of just the `user` then? This way, I believe the `expect()` won’t be needed.